lib/stytch/b2b_client.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require_relative "debug"
+require_relative "b2b_sessions"
+require_relative "b2b_organizations"
+require_relative "b2b_magic_links"
+require_relative "b2b_passwords"
+require_relative "b2b_sso"
+
+module StytchB2B
+  class Client
+    ENVIRONMENTS = %i[live test].freeze
+
+    attr_reader :debug, :organizations, :sessions, :magic_links, :passwords, :sso
+
+    def initialize(project_id:, secret:, env: nil, &block)
+      @api_host = api_host(env, project_id)
+      @project_id = project_id
+      @secret = secret
+
+      create_connection(&block)
+
+      @debug = Stytch::Debug.new(@connection)
+      @organizations = StytchB2B::Organizations.new(@connection)
+      @sessions = StytchB2B::Sessions.new(@connection)
+      @magic_links = StytchB2B::MagicLinks.new(@connection)
+      @passwords = StytchB2B::Passwords.new(@connection)
+      @sso = StytchB2B::SSO.new(@connection)
+    end
+
+    private
+
+    def api_host(env, project_id)
+      case env
+      when :live
+        "https://api.stytch.com"
+      when :test
+        "https://test.stytch.com"
+      when /\Ahttps?:\/\//
+        # If this is a string that looks like a URL, assume it's an internal development URL.
+        env
+      else
+        if project_id.start_with? "project-live-"
+          "https://api.stytch.com"
+        else
+          "https://test.stytch.com"
+        end
+      end
```

Comment:
Now featuring env detection, but I'm curious if there's a cleaner way to do this